feat: successor market (#1500)
vegaprotocol/specs: protocol/0001-MKTF-market_framework.md

}
```

## Successor market

If a market proposal, see [governance](./0028-GOVE-governance.md), designates an existing market as a *parent market* then it must have the same *product*, *settlement asset(s)* and *margin asset(s)*.

> It may propose new risk model and parameters, price monitoring parameters, position and market decimal places and oracles, both for trading terminated and for settlement data.
> Each market can have exactly one market as a *successor* market.
> 
> 1. if there already is a market (possibly pending i.e. in opening auction, see [lifecycle spec](./0043-MKTL-market_lifecycle.md)) naming a parent market which is referenced in the proposal then the proposal is rejected.
> 1. if there are two proposals naming the same parent market then whichever one gets into the pending state first (i.e. passes governance vote) becomes the successor of the named parent; the other proposal is cancelled with reason "parent market not available".

Reviewer: ```suggestion
It may propose a new risk model and parameters, price monitoring parameters, position and market decimal places and oracles, both for trading terminated and for settlement data.
Each market can have exactly one market as a *successor* market.

1. if there already is a market (possibly pending, i.e. in opening auction, see [lifecycle spec](./0043-MKTL-market_lifecycle.md)), naming a parent market, then a subsequent proposal referencing that market is rejected.
1. if there are two proposals naming the same parent market then whichever one gets into the pending state first (i.e. passes governance vote) becomes the successor of the named parent; the other proposal is cancelled with reason "parent market not available".
```

Reviewer: "may propose a new risk model and parameters" - Might this be better phrased "may propose new risk model parameters"?

Author: I am not sure; the meaning I am trying to convey is that the parameters may be new but also the risk model can be different (and if that happens new parameters are needed because different risk models have different parameter sets).

> Some markets can be labelled as *perpetual* at the time of proposal; in this case they cannot serve as a parent market.

Reviewer: And would a market proposal that chooses a perpetual market as its parent then be rejected?

Author: Yes.

> Some markets can be labelled as *perpetual* at the time of proposal; in this case they cannot serve as a parent market.

Reviewer 2: ```suggestion
Some markets can be labelled as *perpetual* (e.g. perpetual futures or spot markets) at the time of proposal; in this case they cannot serve as a parent market.
```

> Some markets can be labelled as *perpetual* (e.g. perpetual futures or spot markets) at the time of proposal; in this case they cannot serve as a parent market.

Reviewer 3: Revisit.

Author: We allow perps and spot to also serve as parent as this will allow us to "reconfigure" those.

> Some markets can be labelled as *perpetual* (e.g. perpetual futures or spot markets) at the time of proposal; in this case they cannot serve as a parent market.

Author: ```suggestion
```
